### PR TITLE
Update shape matching to not match view connections with different views

### PR DESCRIPTION
**Pull request by mmandlis**
https://github.com/PolymerLabs/arcs/issues/590

### CODE REVIEW

**Comment by mmandlis** [comment]
This is somewhat hacky, I'm afraid you aren't going to like it.
But if this is smth we could live for the time being, I'll add test cases to cover it.

File: runtime/recipe/view.js [hunk 1]
```diff
@@ -81,12 +83,18 @@ class View {
...
+  get originalId() { return this._originalId; }
   get recipe() { return this._recipe; }
   get tags() { return this._tags; } // only tags owned by the view
   set tags(tags) { this._tags = tags; }
   get type() { return this._type; } // nullable
   get id() { return this._id; }
-  set id(id) { this._id = id; }
+  set id(id) {
+    if (!this._originalId) {
+      this._originalId = this._id;
```
Comment (dstockwell) [inline runtime/recipe/view.js]: Which strategies can change the id?

File: runtime/recipe/recipe-util.js [hunk 2]
```diff
@@ -94,6 +93,11 @@ class RecipeUtil {
           } else if (forward.has(shapeVC.view) && forward.get(shapeVC.view) !== null) {
             continue;
           }
+          if ((shapeVC.view.fate != 'create' || (recipeVC.view.fate != 'create' && recipeVC.view.originalFate != 'create')) &&
```
Comment (dstockwell) [inline runtime/recipe/recipe-util.js]: Could you add a comment to what where're checking?

File: runtime/strategies/add-use-views.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ export default class AddUseViews extends Strategy {
         if (freeViews.length > 0)
           return;
 
-        var disconnectedConnections = recipe.viewConnections.filter(vc => vc.view == null && !vc.isOptional);
+        var disconnectedConnections = recipe.viewConnections.filter(vc => vc.view == null && !vc.isOptional && vc.name != "descriptions");
```
Comment (dstockwell) [inline runtime/strategies/add-use-views.js]: Needs a comment/todo about descriptions.

File: runtime/recipe/view.js (hunk 1 above)
Comment (mmandlis) [reply runtime/recipe/view.js]: When view fate is "copy", during the recipe instantiation a new view is created and assigned a new ID:
https://github.com/PolymerLabs/arcs/blob/master/runtime/arc.js#L228
with this change, the originalId will keep the ID of the view this one was copied from.

File: runtime/strategies/add-use-views.js (hunk 3 above)
Comment (mmandlis) [reply runtime/strategies/add-use-views.js]: done

File: runtime/recipe/recipe-util.js (hunk 2 above)
Comment (mmandlis) [reply runtime/recipe/recipe-util.js]: done

File: runtime/recipe/view.js (hunk 1 above)
Comment (dstockwell) [reply runtime/recipe/view.js]: Ahh I see. That does seem important to keep around.

Comment (mmandlis) [reply runtime/recipe/view.js]: I know we talked about making a more generic solution for this (back when i added original fate). But for now I just keep piling up hacks and todos...

**Pull request merged**